### PR TITLE
Method getAllByIndex added

### PR DESCRIPTION
**Pull request by thencke**
Hi, this is a tiny modification. I needed a getAllbyIndex, so i implemented it, please merge :D

### CODE REVIEW

File: projects/ngx-indexed-db/src/lib/ngx-indexed-db.ts [hunk 1]
```diff
@@ -196,6 +196,21 @@ export class NgxIndexedDB {
...
+	getAllByIndex(storeName: string, indexName: string, key: any) {
+		return new Promise<any>((resolve, reject) => {
+			this.dbWrapper.validateBeforeTransaction(storeName, reject);
+			let transaction = this.dbWrapper.createTransaction(
+					this.dbWrapper.optionsGenerator(DBMode.readonly, storeName, reject, resolve)
+				),
+				objectStore = transaction.objectStore(storeName),
+				index = objectStore.index(indexName),
+				request = index.getAll(key);
+			request.onsuccess = event => {
```
Comment (assuncaocharles) [inline projects/ngx-indexed-db/src/lib/ngx-indexed-db.ts:209]: Can you handle the rejection also, please?

**Review by assuncaocharles — COMMENTED** [review]
Just one small comment. :)

File: projects/ngx-indexed-db/src/lib/ngx-indexed-db.ts (hunk 1 above)
Comment (thencke) [reply projects/ngx-indexed-db/src/lib/ngx-indexed-db.ts:209]: done :+1:

**Comment by assuncaocharles** [comment]
Great, can you just update the docs for you new method?

**Comment by assuncaocharles** [comment]
Hi @blinkn 

Let me know if you still want to adapt this

**Pull request closed without merging**